docs/md/sink.md

```diff
@@ -0,0 +1,132 @@
+# Sink
+The workload `Sink` models the terminal stage of a processing pipeline. In a 
+typical workload configuration, a `Sink` can be used to write workflow outputs
+to a desired location in the cloud.
+
+## User Guide
+You can configure the type of `Sink` used in your workload by changing the
+`sink` attribute of your workload request.
+
+### Terra Workspace Sink
+You can write workflow outputs to a Terra Workspace using the `Terra Workspace`
+sink. A typical `Terra Workspace` sink configuration in the workload request
+looks like:
+```json
+{
+  "name": "Terra Workspace",
+  "workspace": "{workspace-namespace}/{workspace-name}",
+  "entityType": "{name}",
+  "identifier": "{output-name}",
+  "fromOutputs": {
+    "attribute0": "output0",
+    "attribute1": ["output1", "output2"],
+    "attribute3": "$literal",
+    ...
+  }
+}
+``` 
+The table below summarises the purpose of each attribute in the above request.
+
+| Attribute     | Description                                                  |
+|---------------|--------------------------------------------------------------|
+| `name`        | Selects the `Terra Workspace` sink implementation.           |
+| `workspace`   | The Terra Workspace to write pipeline outputs to.            |
+| `entityType`  | The entity type in the `workspace` to write outputs to.      |
+| `identifier`  | Selects the output that will be used as the entity name.     |
+| `fromOutputs` | Mapping from outputs to attribute names in the `entityType`. |
+
+#### `workspace`
+The workspace is a `"{workspace-namespace}/{workspace-name}"` string as it
+appears in the URL path in the Terra UI. The workspace must exist prior to
+workload creation. You must ensure that `workflow-launcher@firecloud.org` is
+a workspace "Writer" in order to write entities to the workspace.
+
+#### `entityType`
+The `entityType` is the name of the entity type in the workspace that entities
+will be created as. The entity type must exist prior to workload creation and
+must be a table in the workspace.
+
+#### `identifier`
+The `identifier` is the name of a pipeline output that should be used as the 
+name of each newly created entity. 
+
+Example - Let's say the pipeline you're running has an output called
+"sample_name" that uniquely identifies the inputs and outputs to that pipeline.
+By setting `"identifier": "sample_name"` in the sink configuration, entities
+will be created using the "sample_name" as the entity name.
+
+Note that when two sets of pipeline outputs share the same "identifier" value,
```

Comment:
Is this section worth flagging as a warning?

Stepping outside of documentation, do we have sufficient visibility into an identifier collision to flag as a `UserException`?

Reply:
This is by design

Reply:
Ah right, to smooth retries.